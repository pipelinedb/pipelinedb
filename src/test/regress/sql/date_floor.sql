-- nulls
SELECT date_floor(NULL, NULL);
SELECT date_floor(NULL, INTERVAL '5 minute');
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', NULL);

-- valid floors
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '15 second');
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '10 minute');
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '3 hour');
SELECT date_floor(TIMESTAMP '2001-02-18 20:38:40', INTERVAL '5 day');
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '2 month');
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '7 year');

-- invalid floors
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '150 second');
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '65 minute');
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '48 hour');
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '36 day');
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '18 month');
