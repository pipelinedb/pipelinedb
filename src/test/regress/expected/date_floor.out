-- nulls
SELECT date_floor(NULL, NULL);
 date_floor 
------------
 
(1 row)

SELECT date_floor(NULL, INTERVAL '5 minute');
 date_floor 
------------
 
(1 row)

SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', NULL);
          date_floor          
------------------------------
 Fri Feb 16 20:38:40 2001 PST
(1 row)

-- valid floors
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '15 second');
          date_floor          
------------------------------
 Fri Feb 16 20:38:30 2001 PST
(1 row)

SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '10 minute');
          date_floor          
------------------------------
 Fri Feb 16 20:30:00 2001 PST
(1 row)

SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '3 hour');
          date_floor          
------------------------------
 Fri Feb 16 18:00:00 2001 PST
(1 row)

SELECT date_floor(TIMESTAMP '2001-02-18 20:38:40', INTERVAL '5 day');
          date_floor          
------------------------------
 Fri Feb 16 00:00:00 2001 PST
(1 row)

SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '2 month');
          date_floor          
------------------------------
 Mon Jan 01 00:00:00 2001 PST
(1 row)

SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '7 year');
          date_floor          
------------------------------
 Sun Jan 01 00:00:00 1995 PST
(1 row)

-- invalid floors
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '150 second');
ERROR:  floor interval must span only a single unit
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '65 minute');
ERROR:  floor interval must span only a single unit
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '48 hour');
ERROR:  floor interval must span only a single unit
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '36 day');
ERROR:  floor interval must span only a single unit
SELECT date_floor(TIMESTAMP '2001-02-16 20:38:40', INTERVAL '18 month');
ERROR:  floor interval must span only a single unit
